src/librbd/internal.cc

```diff
@@ -531,7 +531,10 @@ int remove_object_map(ImageCtx *ictx) {
       }
 
       r = ictx->image_watcher->notify_snap_create(snap_name);
-      if (r == -EEXIST) {
+      if (r == 0) {
+        notify_change(ictx->md_ctx, ictx->header_oid, ictx);
+        return 0;
+      } else if (r == -EEXIST) {
         return 0;
```

Comment:
Good point -- the EEXIST case should also send out the notification to catch timeout recoveries.